app/javascript/packages/document-capture/components/address-search.tsx

```diff
@@ -35,6 +35,7 @@ function AddressSearch({
   registerField = () => undefined,
 }: AddressSearchProps) {
   const validatedFieldRef = useRef<HTMLFormElement | null>(null);
+  const [noAddressFoundErrors, setNoAddressFoundErrors] = useState('');
```

Comment:
`noAddressFoundErrors` appears to have the sole effect of ensuring that the `useEffect` below runs twice every time the address candidates change between `null`/zero and 1+ length. Does that yield a specific result that we need? If so, then is there a more straightforward way to represent the operation?

Reply:
I refactored and removed the `noAddressFoundErrors`.  My reasoning was that
a) it's not necessary for the `useEffect` hook, because that can use `addressCandidates` as a dependency
b) the state variable I was setting, `'Not a real address'`, wasn't being used anywhere in the code